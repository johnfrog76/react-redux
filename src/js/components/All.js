
import React from "react";
import {ProgressBar} from 'react-bootstrap';

class LoadingSpinner extends React.Component {
    constructor() {
        super();
    }
    render() {
        return (
            <div id="loading">
              <i className="fa fa-cog fa-spin"></i> loading...
            </div>
        )
    }
}

class ErrorComponent extends React.Component {
    constructor() {
        super();
    }
    render() {
        let data = this.props.data;
        return (
            <div class="error">
                <i class="fa fa-warning"></i> {data}
            </div>
        )
    }
}

class Portfolio extends React.Component {
    constructor() {
        super();
    }
    render() {
        let data = this.props.data;
        return (
            <div className="portfolio">
                {
                    data.map((item, i) => {
                        let headingID = 'heading' + item.id;
                        let photos = item.images.map(function(image, n) {
                            let str = 'http://sbtest.com/portfolio/images/' + image.img;
                            return (
                                <div key={n} className="photo-item">
                                    <img src={str} />
                                    <figcaption>{image.caption}</figcaption>
                                </div>
                            )
                        });


                        return (
                            <div key={i}>
                                <h2 id={headingID}>{item.name}</h2>
                                <div>{photos}</div>
                            </div>
                        );
                    })
                }
            </div>
        )
    }
}

class LoginForm extends React.Component {
    constructor() {
        super();
        this.handleSubmit = this._handleSubmit.bind(this);
    }
    _handleSubmit (e) {
        e.preventDefault();
        let user = e.target.elements.user.value.trim();
        let pass = e.target.elements.pass.value.trim();

        if (!user || !pass) {
            return;
        }

        return this.props.layout.fetchSession({user: user, pass: pass});
    }
    render () {
        let data = this.props.data;
        return (
            <form className="MessageForm" onSubmit={this.handleSubmit}>
            <input type="hidden" value="prayer" />
              <input
                type="text"
                className="form-control"
                name="user"
                placeholder="user"
                autoComplete="off"
                value={this.user}
                onChange={this.handleNameChange}
              />
              <input
                type="password"
                className="form-control"
                name="pass"
                placeholder="pass"
                autoComplete="off"
                value={this.pass}
                onChange={this.handlePassChange}
              />
              <button
                type="submit"
                className="btn btn-primary"
              >
                Log In
              </button>
            </form>
        )
    }
}

class RecipeTable extends React.Component {
    constructor() {
        super();
        this.handleClick = this._handleClick.bind(this);
    }
    _handleClick (evt) {
        evt.preventDefault();
        this.props.layout.onViewRecipeDetail(parseFloat(evt.target.id));
        evt.target.blur();
    }
    render() {
        let data = this.props.data;

        return (
            <div class="table-wrap">
                <table class="table">
                    <thead>
                    <tr>
                        <th>Category</th>
                        <th>Rating</th>
                        <th>Name</th>
                        <th>ID</th>
                        <th>Details</th>
                    </tr>
                    </thead>
                    <tbody>
                    {
                        data.map((item, i) => {
                            return (
                                <tr key={i}>
                                    <td>{item.category}</td>
                                    <td>{item.rating}</td>
                                    <td>{item.r_name}</td>
                                    <td>{item.id}</td>
                                    <td><a id={item.id}
                                            href="#"
                                            onClick={this.handleClick}>View
                                        </a>
                                    </td>
                                </tr>
                            );
                        })
                    }
                    </tbody>
                </table>
            </div>
        )
    }
}

class RecipeViewDetails extends React.Component {
    constructor() {
        super();
        this.handleClick = this._handleClick.bind(this);
    }
    _handleClick(e) {
        e.preventDefault();
        this.props.layout.fetchRecipes(e);
        e.target.blur();
    }
    render() {
        const data = this.props.data.recipe;
        let comments = (data.comments.length < 1) ? [] : data.comments;
        let ingredients = data.ingredients;
        let steps = data.steps;
        return (
            <div className="recipeItem">
            <p><a href="#" onClick={this.handleClick}>Back &gt;</a></p>
            <h3>{data.r_name}</h3>
                <h5>Ingredients</h5>
                <ul>
                    {
                    ingredients.map((item, i) => {
                            return (
                                <li key={'ingredient' + i}>{item}</li>
                            );
                        })
                    }
                </ul>
                <h5>Steps</h5>
                <ul>
                    {
                        steps.map((item, i) => {
                            return (
                                <li key={'step' + i}>{item}</li>
                            );
                        })
                    }
                </ul>
                <h5>Comments</h5>
                <ul>
                    {
                        comments.map((item, i) => {
                            return (
                                <li key={'comment' + i}>
                                    {item.user}: {item.comment}
                                </li>
                            );
                        })
                    }
                </ul>
            </div>
        );
    }
}

class ViewSongs extends React.Component {
    constructor() {
        super();
        this.handlePlay = this._handlePlay.bind(this);
        this.handlePause = this._handlePause.bind(this);
        this.pauseAll = this._pauseAll.bind(this);
        this.increment = this._increment.bind(this);
        this.setInt = this._setInt.bind(this);
        this.clearInt = this._clearInt.bind(this);

    }
    _increment(ele, progEle, pos) {
        let p = Math.round(ele.currentTime / ele.duration * 100);
        let retObj = {
            pos: pos,
            playPos: p,
            playing: true
        };

        if (p < 100) {
            this.props.layout.updateSong(retObj);
        } else {
            retObj.playPos = 0;
            retObj.playing = false;
            this.props.layout.updateSong(retObj);
            this.clearInt();
        }
    }
    _clearInt() {
        if (this.myInterval) {
            clearInterval(this.myInterval);
        }
    }
    _setInt(ele, progEle, pos) {
        let self = this;
        this.clearInt();
        this.myInterval = setInterval(function(){ self.increment(ele, progEle, pos) }, 500);
    }
    _pauseAll() {
        let i;
        let items = document.getElementsByTagName('audio')
        let len = items.length;

        for (i = 0; i < len; i++) {
            items[i].pause();
        }

        this.props.layout.pauseAllSongs();
    }
    _handlePlay(e) {
        let ele = e.target.parentElement.firstElementChild;
        let progEle = e.target.previousElementSibling.firstChild;
        let pos = parseFloat(e.target.id.replace('play', ''));
        let retObj = {
            pos: pos,
            playPos: 0,
            playing: true
        };
        e.preventDefault();
        e.target.blur();
        this.pauseAll();
        ele.play();
        this.setInt(ele, progEle, pos);
        this.props.layout.updateSong(retObj);
    }
    _handlePause(e) {
        let pos = parseFloat(e.target.id.replace('pause', ''));
        let retObj = {
            pos: pos,
            playing: false
        };
        e.preventDefault();
        e.target.blur();
        e.target.parentElement.firstElementChild.pause();
        this.clearInt();
        this.props.layout.updateSong(retObj);
    }
    render() {
        const data = this.props.data;

        return (
            <div className="songs">
                <h5>Songs</h5>
                <ul>
                    {
                        data.map((item, i) => {
                            const mySrc = 'http://katie.sbtest.com/upload/' + item.audioclip;
                            let playClass = item.playing ? 'hidden' : 'btn btn-xs fa fa-play';
                            let pauseClass = item.playing ? 'btn btn-xs fa fa-pause' : 'hidden';

                            return (
                                <li key={'song' + i}>
                                    <audio id={'audio' + item.id} preload="auto">
                                        <source
                                            src={mySrc}
                                            type="audio/mpeg"
                                        />
                                            Your browser does not support the audio element.
                                    </audio>
                                    <div className="song-title">{item.title}</div>
                                    <ProgressBar id={'progress' + item.id} now={item.playPos} />
                                    <button
                                        id={'play' + i}
                                        onClick={this.handlePlay}
                                        className={playClass}
                                    >
                                    </button>
                                    <button
                                        id={'pause' + i}
                                        onClick={this.handlePause}
                                        className={pauseClass}
                                    >
                                    </button>
                                </li>
                            );
                        })
                    }
                </ul>
            </div>
        );
    }
}


class FiddlesJS extends React.Component {
    constructor() {
        super();
    }
    render() {
        let data = this.props.data;

        return (
            <div className="fiddle">
                {
                    data.map((item, i) => {
                        let colors = ['#3F51B5','#4CAF50', '#009688', '#00BCD4'];
                        let pageIndex = i % colors.length;
                        let styleColor = {
                            color: colors[pageIndex]
                        };
                        let backgroundStyle = {
                            backgroundColor: colors[pageIndex],
                            padding: '10px 10px 5px 10px',
                            marginBottom: '2.5rem'
                        };
                        let groupClass = 'group' + i;
                        let headingID = 'heading' + item.id;
                        let heading = item.fiddles.length === 0 ? null :
                            <h2 id={headingID} style={styleColor}>{item.name}</h2>;
                        let fiddles = item.fiddles.map(function(fiddle, n) {
                            let str = '//jsfiddle.net/' + fiddle.fiddle +
                                '/embedded/result,js,html,css/dark/';
                            return (
                                <div key={n} className="fiddle-item" style={backgroundStyle}>
                                    <h3>{fiddle.title}</h3>
                                    <p>{fiddle.description}</p>
                                    <div className="fiddle-wrap">
                                        <iframe
                                            width="100%"
                                            height="500"
                                            src={str}
                                            scrolling="yes"
                                        >
                                        </iframe>
                                    </div>
                                </div>
                            )
                        });

                        return (
                            <div key={i} className={groupClass}>
                                {heading}
                                <div>{fiddles}</div>
                            </div>
                        );
                    })
                }
            </div>
        )
    }
}

export {
    LoadingSpinner,
    Portfolio,
    FiddlesJS,
    LoginForm,
    RecipeTable,
    RecipeViewDetails,
    ErrorComponent,
    ViewSongs
}
