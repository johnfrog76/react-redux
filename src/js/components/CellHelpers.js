
import React from 'react'
import {Cell} from 'fixed-data-table-2'

// helper for fixed-data-table-2
class SortHeaderCell extends React.Component {
  constructor (props) {
    super(props)

    this.sortTypes = {
      ASC: 'ASC',
      DESC: 'DESC'
    }

    this.onCellSort = this._onCellSort.bind(this)
    this.reverseSortDirection = this._reverseSortDirection.bind(this)
  }

  render () {
    let {onCellSort, children, ...props} = this.props
    return (
      <Cell>
        <a onClick={this.onCellSort}>
          <span>{children} {props.sortDir ? (props.sortDir === this.sortTypes.DESC ? '↓' : '↑') : ''}</span>
        </a>
      </Cell>
    )
  }

  _reverseSortDirection (sortDir) {
    return sortDir === this.sortTypes.DESC ? this.sortTypes.ASC : this.sortTypes.DESC
  }

  _onCellSort (e) {
    e.preventDefault()

    if (this.props.onSortChange) {
      this.props.onSortChange(
        this.props.columnKey,
        this.props.sortDir ? this._reverseSortDirection(this.props.sortDir)
            : this.sortTypes.DESC
      )
    }
  }
}

// RatingCell is a helper for fixed-data-table-2
class RatingCell extends React.Component {
  constructor () {
    super()
  }
  render () {
    const {rowIndex, field, data, ...props} = this.props
    let rating = data[rowIndex].rating
    let ret = ''
    switch (rating) {
      case 1: {
        ret = <div className='rating'>
          <em className='fa fa-star' />
        </div>
        break
      }
      case 2: {
        ret = <div className='rating'>
          <em className='fa fa-star' />
          <em className='fa fa-star' />
        </div>
        break
      }
      case 3: {
        ret = <div className='rating'>
          <em className='fa fa-star' />
          <em className='fa fa-star' />
          <em className='fa fa-star' />
        </div>
        break
      }
      case 4: {
        ret = <div className='rating'>
          <em className='fa fa-star' />
          <em className='fa fa-star' />
          <em className='fa fa-star' />
          <em className='fa fa-star' />
        </div>
        break
      }
    }
    return (
      <Cell {...props}>{ret}</Cell>
    )
  }
}

// TextCell is a helper for fixed-data-table-2
class TextCell extends React.Component {
  constructor () {
    super()
  }
  render () {
    const {rowIndex, field, data, ...props} = this.props
    const columnKey = this.props.columnKey
    return (
      <Cell {...props}>{data[rowIndex][columnKey]}</Cell>
    )
  }
}

// AcitonCell is a helper for fixed-data-table-2
class ActionCell extends React.Component {
  constructor () {
    super()
  }
  render () {
    const {rowIndex, field, data, ...props} = this.props
    let myId = data[rowIndex].id
    let cellContents = (myId === '') ? ''
      : <a id={myId} href='#' onClick={props.action.handleClick}>View</a>

    return (
      <Cell {...props}>{cellContents}</Cell>
    )
  }
}

export {
  TextCell,
  ActionCell,
  SortHeaderCell,
  RatingCell
}
