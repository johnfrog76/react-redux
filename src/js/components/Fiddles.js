import React from 'react'

class FiddlesJS extends React.Component {
  constructor () {
    super()
  }
  render () {
    let data = this.props.data

    return (
      <div>
        <h1>Fiddles</h1>
        <div className='fiddle'>
          {
            data.map((item, i) => {
              let colors = ['#AFB42B', '#4CAF50', '#009688', '#00BCD4']
              let pageIndex = i % colors.length
              let styleColor = {
                color: colors[pageIndex]
              }
              let backgroundStyle = {
                backgroundColor: colors[pageIndex],
                padding: '10px 10px 5px 10px',
                marginBottom: '2.5rem'
              }
              let groupClass = 'group' + i
              let headingID = 'heading' + item.id
              let heading = item.fiddles.length === 0 ? null
                  : <h2 id={headingID} style={styleColor}>{item.name}</h2>
              let fiddles = item.fiddles.map(function (fiddle, n) {
                let str = '//jsfiddle.net/' + fiddle.fiddle +
                '/embedded/result,js,html,css/dark/'
                return (
                  <div key={n} className='fiddle-item' style={backgroundStyle}>
                    <h3>{fiddle.title}</h3>
                    <p>{fiddle.description}</p>
                    <div className='fiddle-wrap'>
                      <iframe
                        width='100%'
                        height='500'
                        src={str}
                        scrolling='yes'
                       />
                    </div>
                  </div>
                )
              })

              return (
                <div key={i} className={groupClass}>
                  {heading}
                  <div>{fiddles}</div>
                </div>
              )
            })
          }
        </div>
      </div>
    )
  }
}

export {
  FiddlesJS
}
