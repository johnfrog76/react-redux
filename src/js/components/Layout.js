import React from 'react'
import { connect } from 'react-redux'
import _ from 'lodash'

import {
  Button,
  Navbar,
  Nav,
  NavItem
} from 'react-bootstrap'

import {
  fetchPortfolio,
  displayPortfolio,
  clearPortfolio,
  inactivePortfolio
} from '../actions/portfolioActions'

import {
  fetchFiddles,
  displayFiddles,
  clearFiddles,
  inactiveFiddles
} from '../actions/fiddleActions'

import {
  displayAbout,
  inactiveAbout
} from '../actions/aboutActions'

import {
  fetchSession,
  showLoginForm,
  clearSession
} from '../actions/recipeSessionActions'

import {
  fetchRecipe,
  clearRecipe,
  inactiveRecipe
} from '../actions/recipeItemActions'

import {
  fetchRecipes,
  displayRecipes,
  clearRecipes,
  inactiveRecipes
} from '../actions/recipeActions'

import {
  fetchSongs,
  displaySongs,
  clearSongs,
  inactiveSongs,
  updateSong,
  pauseAllSongs
} from '../actions/songsActions'

import {
  LoadingSpinner,
  AboutUs,
  LoginForm,
  RecipeTable,
  ResponsiveGrid,
  RecipeViewDetails,
  ErrorComponent,
  FooterCont
} from '../components/All'

import {
  FiddlesJS
} from '../components/Fiddles'

import {
  Portfolio
} from '../components/Portfolio'

import {
  ViewSongs
} from '../components/ViewSongs'

@connect((store) => {
  return {
    portfolio: store.portfolio,
    fiddles: store.fiddles,
    session: store.session,
    recipes: store.recipes,
    recipe: store.recipe,
    songs: store.songs,
    about: store.about
  }
})

export default class Layout extends React.Component {
  componentWillMount () {
    // fetch session here
    this.props.dispatch(displayAbout())
    this.props.dispatch(fetchSession({user: '1', pass: 'a'}))
  }

  needsCache (key) {
    // return true if needs cache refresh or null
    const life = 1000 * 60 * 1
    let myObj = this.props[key]
    let ret = true; // assume needed

    if (myObj.cache) {
      // cache exists
      let d = new Date();
      let millis = d.getTime()
      let diff = (millis - myObj.cache) - life

      ret =  diff > 0 ? true : false; // gtn 0 refresh
    }
    return ret
  }

  clearStore () {
    this.props.dispatch(clearRecipe())
    this.props.dispatch(clearRecipes())
    this.props.dispatch(clearFiddles())
    this.props.dispatch(clearPortfolio())
    this.props.dispatch(clearSongs())
  }

  clearTabs () {
    this.props.dispatch(inactiveAbout())
    this.props.dispatch(inactiveFiddles())
    this.props.dispatch(inactivePortfolio())
    this.props.dispatch(inactiveRecipes())
    this.props.dispatch(inactiveRecipe())
    this.props.dispatch(inactiveSongs())
  }

  fetchPortfolio (e) {
    e.preventDefault()
    e.target.blur()
    this.clearTabs()

    if (this.needsCache('portfolio')) {
      this.props.dispatch(fetchPortfolio())
    } else {
      this.props.dispatch(displayPortfolio())
    }
  }

  showLoginForm (e) {
    e.preventDefault()
    e.target.blur()
    this.clearTabs()

    if (!_.isEmpty(this.props.session.sess)) {
      if (this.needsCache('recipes')) {
        this.props.dispatch(fetchRecipes())
      } else {
        this.props.dispatch(displayRecipes())
      }
    } else {
      this.props.dispatch(showLoginForm())
    }
  }

  updateSong (song) {
    this.props.dispatch(updateSong(song))
  }

  pauseAllSongs () {
    this.props.dispatch(pauseAllSongs())
  }

  fetchAbout (e) {
    e.preventDefault()
    e.target.blur()
    this.clearTabs()
    this.props.dispatch(displayAbout())
  }

  fetchRecipes (e) {
    e.preventDefault()
    e.target.blur()
    this.clearTabs()

    if (this.needsCache('recipes')) {
      this.props.dispatch(fetchRecipes())
    } else {
      this.props.dispatch(displayRecipes())
    }
  }

  fetchSession (obj) {
    this.clearTabs()
    this.props.dispatch(fetchRecipes())
  }

  fetchFiddles (e) {
    e.preventDefault()
    e.target.blur()
    this.clearTabs()
    if (this.needsCache('fiddles')) {
      this.props.dispatch(fetchFiddles())
    } else {
      this.props.dispatch(displayFiddles())
    }
  }

  fetchSongs (e) {
    e.preventDefault()
    e.target.blur()
    this.clearTabs()

    if (this.needsCache('songs')) {
      this.props.dispatch(fetchSongs())
    } else {
      this.props.dispatch(displaySongs())
    }
  }

  onViewRecipeDetail (recipeToShowID) {
    this.props.dispatch(fetchRecipe(recipeToShowID))
  }

  isActive (value) {
    return value === true ? 'active' : ''
  }

  render () {
    const {portfolio, fiddles, recipes, recipe, session, songs, about} = this.props

    let tab = <LoadingSpinner />

    if (fiddles.isActive) {
      tab = <FiddlesJS data={fiddles.fiddleitems} />
    } else if (fiddles.fetching) {
      tab = <LoadingSpinner />
    } else if (about.isActive) {
      tab = <AboutUs />
    } else if (portfolio.isActive) {
      tab = <Portfolio data={portfolio.portfolioitems} />
    } else if (portfolio.error) {
      tab = <ErrorComponent data={portfolio.error.message} />
    } else if (fiddles.error) {
      tab = <ErrorComponent data={fiddles.error.message} />
    } else if (songs.error) {
      tab = <ErrorComponent data={songs.error.message} />
    } else if (songs.fetching) {
      tab = <LoadingSpinner />
    } else if (songs.isActive) {
      tab = <ViewSongs layout={this} data={songs.songitems} />
    } else if (session.error) {
        tab = <ErrorComponent data={session.error} />
      } else if (session.showForm === true) {
          tab = <LoginForm layout={this} data={session} />
        } else if (!_.isEmpty(session.sess) && session.showForm === false) {
          if (recipe.isActive) {
            tab = <RecipeViewDetails
              layout={this}
              data={recipe}
            />
          } else if (recipe.error) {
            tab = <ErrorComponent data={recipe.error} />
          } else if (recipes.isActive) {
            tab = <ResponsiveGrid data={recipes.recipeitems}
              layout={this}
              onViewRecipeDetail={this.onViewRecipeDetail}
            />
          }
        }

    return <div>
      <Navbar inverse collapseOnSelect>
        <Navbar.Header>
          <Navbar.Toggle />
          <Navbar.Brand>
            <a
              href='#'
              onClick={this.fetchAbout.bind(this)}
            >JW
            </a>
          </Navbar.Brand>
        </Navbar.Header>
        <Navbar.Collapse>
          <Nav>
            <NavItem
              eventKey={1}
              className={this.isActive(this.props.about.isActive)}
              onClick={this.fetchAbout.bind(this)}
              href='#'
            >About
            </NavItem>
            <NavItem
              eventKey={2}
              className={this.isActive(this.props.portfolio.isActive)}
              onClick={this.fetchPortfolio.bind(this)}
              href='#'
            >Portfolio
            </NavItem>
            <NavItem
              eventKey={3}
              className={this.isActive(this.props.fiddles.isActive)}
              onClick={this.fetchFiddles.bind(this)}
              href='#'
            >Fiddles
            </NavItem>
            <NavItem
              eventKey={4}
              className={this.isActive(this.props.recipes.isActive)}
              onClick={this.showLoginForm.bind(this)}
              href='#'
            >Recipes
            </NavItem>
            <NavItem
              eventKey={5}
              className={this.isActive(this.props.songs.isActive)}
              onClick={this.fetchSongs.bind(this)}
              href='#'
            >Songs
            </NavItem>
          </Nav>
        </Navbar.Collapse>
      </Navbar>
      <div className='container'>
        <div className='row'>
          <div className='col-sm-12 main-content-area'>
            {tab}
          </div>
        </div>
      </div>
      <div className='footer'>
        <FooterCont />
      </div>
    </div>
  }
}
