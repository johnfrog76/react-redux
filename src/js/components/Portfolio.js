import React from 'react'

class Portfolio extends React.Component {
  constructor () {
    super()
  }
  render () {
    let data = this.props.data
    return (
      <div>
        <h1>Portfolio</h1>
        <div className='portfolio'>
          {
            data.map((item, i) => {
              let headingID = 'heading' + item.id
              let photos = item.images.map(function (image, n) {
                let str = 'http://sbtest.com/portfolio/images/' + image.img
                return (
                  <div key={n} className='photo-item'>
                    <img src={str} />
                    <figcaption>{image.caption}</figcaption>
                  </div>
                )
              })

              return (
                <div key={i}>
                  <h2 id={headingID}>{item.name}</h2>
                  <div>{photos}</div>
                </div>
              )
            })
          }
        </div>
      </div>
    )
  }
}

export {
  Portfolio
}
