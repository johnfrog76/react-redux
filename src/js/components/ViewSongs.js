import React from 'react'
import {ProgressBar} from 'react-bootstrap'

class ViewSongs extends React.Component {
  constructor () {
    super()
    this.handlePlay = this._handlePlay.bind(this)
    this.handlePause = this._handlePause.bind(this)
    this.pauseAll = this._pauseAll.bind(this)
    this.increment = this._increment.bind(this)
    this.setInt = this._setInt.bind(this)
    this.clearInt = this._clearInt.bind(this)
  }
  _increment (ele, progEle, pos) {
    let p = Math.round(ele.currentTime / ele.duration * 100)
    let retObj = {
      pos: pos,
      playPos: p,
      playing: true
    }

    if (p < 100) {
      this.props.layout.updateSong(retObj)
    } else {
      retObj.playPos = 0
      retObj.playing = false
      this.props.layout.updateSong(retObj)
      this.clearInt()
    }
  }
  _clearInt () {
    if (this.myInterval) {
      clearInterval(this.myInterval)
    }
  }
  _setInt (ele, progEle, pos) {
    let self = this
    this.clearInt()
    this.myInterval = setInterval(function () {
        // TODO: do this another way
      if (document.getElementById(ele.id)) {
        self.increment(ele, progEle, pos)
      } else {
        self.clearInt()
      }
    }, 1000)
  }
  _pauseAll () {
    let i
    let items = document.getElementsByTagName('audio')
    let len = items.length

    for (i = 0; i < len; i++) {
      items[i].pause()
    }

    this.props.layout.pauseAllSongs()
  }
  _handlePlay (e) {
    let ele = e.target.parentElement.firstElementChild
    let progEle = e.target.previousElementSibling.firstChild
    let pos = parseFloat(e.target.id.replace('play', ''))
    let retObj = {
      pos: pos,
      playPos: 0,
      playing: true
    }
    e.preventDefault()
    e.target.blur()
    this.pauseAll()
    ele.play()
    this.setInt(ele, progEle, pos)
    this.props.layout.updateSong(retObj)
  }
  _handlePause (e) {
    let pos = parseFloat(e.target.id.replace('pause', ''))
    let retObj = {
      pos: pos,
      playing: false
    }
    e.preventDefault()
    e.target.blur()
    e.target.parentElement.firstElementChild.pause()
    this.clearInt()
    this.props.layout.updateSong(retObj)
  }
  render () {
    const data = this.props.data

    return (
      <div className='songs'>
        <h1>Songs</h1>
        <ul>
          {
            data.map((item, i) => {
              const mySrc = 'http://katie.sbtest.com/upload/' + item.audioclip
              let playClass = item.playing ? 'hidden' : 'btn btn-xs fa fa-play'
              let pauseClass = item.playing ? 'btn btn-xs fa fa-pause' : 'hidden'

              return (
                <li key={'song' + i}>
                  <audio id={'audio' + item.id} preload='auto'>
                    <source
                      src={mySrc}
                      type='audio/mpeg'
                      />
                      Your browser does not support the audio element.
                  </audio>
                  <div className='song-title'>{item.title}</div>
                  <ProgressBar id={'progress' + item.id} now={item.playPos} />
                  <button
                    id={'play' + i}
                    onClick={this.handlePlay}
                    className={playClass}
                  />
                  <button
                    id={'pause' + i}
                    onClick={this.handlePause}
                    className={pauseClass}
                  />
                </li>
              )
            })
          }
        </ul>
      </div>
    )
  }
}

export {
  ViewSongs
}
