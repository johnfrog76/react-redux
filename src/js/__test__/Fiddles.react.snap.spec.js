/* global test expect */
import React from 'react'
import {FiddlesJS} from '../components/Fiddles'
import renderer from 'react-test-renderer'

test('Testing Snapshot of FiddlesJS component', () => {
  const mockData = [{
    'id': 1,
    'name': 'Data Structures',
    'fiddles': [{
      'id': 1,
      'fiddle': '3e1753dw',
      'title': 'Singly Linked List',
      'description': 'Add, List and Remove items from linked list.',
      'active': true,
      'order': 1,
      'cat_id': 1
    }]
  }]

  const component = renderer.create(<FiddlesJS data={mockData} />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
