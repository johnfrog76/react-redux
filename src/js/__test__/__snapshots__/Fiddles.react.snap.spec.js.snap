// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Snapshot of FiddlesJS component 1`] = `
<div>
  <h1>
    Fiddles
  </h1>
  <div
    className="fiddle"
  >
    <div
      className="group0"
    >
      <h2
        id="heading1"
        style={
          Object {
            "color": "#AFB42B",
          }
        }
      >
        Data Structures
      </h2>
      <div>
        <div
          className="fiddle-item"
          style={
            Object {
              "backgroundColor": "#AFB42B",
              "marginBottom": "2.5rem",
              "padding": "10px 10px 5px 10px",
            }
          }
        >
          <h3>
            Singly Linked List
          </h3>
          <p>
            Add, List and Remove items from linked list.
          </p>
          <div
            className="fiddle-wrap"
          >
            <iframe
              height="500"
              scrolling="yes"
              src="//jsfiddle.net/3e1753dw/embedded/result,js,html,css/dark/"
              width="100%"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
